prediction_market_agent_tooling/deploy/betting_strategy.py

```diff
@@ -176,8 +226,68 @@ def calculate_trades(
         )
         return trades
 
+    def calculate_price_impact_for_bet_amount(
+        self, buy_direction: bool, bet_amount: float, yes: float, no: float, fee: float
+    ) -> float:
+        total_outcome_tokens = yes + no
+        expected_price = (
+            no / total_outcome_tokens if buy_direction else yes / total_outcome_tokens
+        )
+
+        tokens_to_buy = get_buy_outcome_token_amount(
+            bet_amount, buy_direction, yes, no, fee
+        )
+
+        actual_price = bet_amount / tokens_to_buy
+        # price_impact should always be > 0
+        price_impact = (actual_price - expected_price) / expected_price
+        return price_impact
+
+    def calculate_bet_amount_for_price_impact(
+        self,
+        market: AgentMarket,
+        kelly_bet: SimpleBet,
+        fee: float,
+    ) -> float:
+        def calculate_price_impact_deviation_from_target_price_impact(
+            bet_amount: xDai,
+        ) -> float:
+            price_impact = self.calculate_price_impact_for_bet_amount(
+                kelly_bet.direction,
+                bet_amount,
+                yes_outcome_pool_size,
+                no_outcome_pool_size,
+                fee,
+            )
+            # We return abs for the algorithm to converge to 0 instead of the min (and possibly negative) value.
+
+            max_price_impact = check_not_none(self.max_price_impact)
+            return abs(price_impact - max_price_impact)
+
+        if not market.outcome_token_pool:
+            logger.warning(
+                "Market outcome_token_pool is None, cannot calculate bet amount"
+            )
+            return kelly_bet.size
+
+        yes_outcome_pool_size = market.outcome_token_pool[
+            market.get_outcome_str_from_bool(True)
+        ]
+        no_outcome_pool_size = market.outcome_token_pool[
+            market.get_outcome_str_from_bool(False)
+        ]
+
+        optimized_bet_amount = minimize_scalar(
+            calculate_price_impact_deviation_from_target_price_impact,
+            bounds=(min(yes_outcome_pool_size, no_outcome_pool_size) / 1000, 1000),
```

Comment:
I see you get some exception with max bound of `sys.maxsize`. A simple way of making it more robust would be just to set the upper bound relative to total pool size.

I tried
```suggestion
            bounds=(0, 1000 * (yes_outcome_pool_size + no_outcome_pool_size)),
```

and it seems to work fine, and is more robust imo.

Reply:
Works for me as well. Implemented.